Doc/library/warnings.rst

```diff
@@ -214,6 +214,18 @@ Some examples::
    ignore,default:::mymodule    # Only report warnings triggered by "mymodule"
    error:::mymodule             # Convert warnings to errors in "mymodule"
 
+Note that :func:`filterwarnings` filters have subtle differences from :option:`-W` and :envvar:`PYTHONWARNINGS`::
+
+   filterwarnings("ignore", message=".*generic", module=r"yourmodule\.submodule")
+      # Ignore warnings in "yourmodule.submodule" which contain "generic"
+   filterwarnings("ignore", module="yourmodule.*")
```

Comment:
```suggestion
   filterwarnings("ignore", module="yourmodule\.*")
```
I think this should be escaped too.

Reply:
Also, instead of using comments, would it be clearer (formatting wise) to use multiple code blocks?

Reply:
Ditto: https://github.com/python/cpython/pull/106618/files#r1261710032